src/common/transformations/src/transformations/common_optimizations/move_eltwise_up_data_movement.cpp

```diff
@@ -126,3 +130,93 @@ ov::pass::MoveEltwiseUpThroughDataMov::MoveEltwiseUpThroughDataMov(
+        if (!constant_op)
+            return false;
+
+        if (output.get_target_inputs().size() != 1)
+            return false;
+
+        const auto& shape = constant_op->get_shape();
+        return std::count_if(shape.begin(), shape.end(), [](size_t v) {
+                   return v > 1;
+               }) == 1;
+    };
+
+    auto eltw_predicate = [](const ov::Output<ov::Node>& output) {
+        if (output.get_target_inputs().size() != 1)
+            return false;
+
+        auto node = output.get_node();
+
+        if (node->get_input_partial_shape(0).size() < node->get_input_partial_shape(1).size())
+            return false;
+
+        return true;
+    };
+
+    auto eltw_data_flow_in =
+        ov::pass::pattern::wrap_type<ov::op::v1::Reshape, ov::op::v0::Squeeze, ov::op::v0::Unsqueeze>();
+    auto eltw_const_in = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(const_predicate);
+    auto eltwise_pattern =
+        ov::pass::pattern::wrap_type<ov::op::util::BinaryElementwiseArithmetic>({eltw_data_flow_in, eltw_const_in},
+                                                                                eltw_predicate);
+
+    ov::matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto eltwise = pattern_map.at(eltwise_pattern).get_node_shared_ptr();
+        if (transformation_callback(eltwise)) {
+            return false;
+        }
+
+        auto const_shape = eltwise->get_input_shape(1);
+        size_t channel_idx = 0;
+        size_t channel_val = 0;
+        for (size_t i = 0; i < const_shape.size(); i++) {
+            if (const_shape[i] > 1) {
+                channel_idx = i;
+                channel_val = const_shape[i];
+            }
+        }
+
+        auto parent = pattern_map.at(eltw_data_flow_in).get_node_shared_ptr();
+        const auto& parent_in_pshape = parent->get_input_partial_shape(0);
+        auto paranet_in_channel_dim =
+            parent_in_pshape.size() <= channel_idx ? ov::Dimension(1) : parent_in_pshape[channel_idx];
+        auto paranet_out_channel_dim = parent->get_output_partial_shape(0)[channel_idx];
+        if (paranet_in_channel_dim.is_dynamic() || paranet_in_channel_dim != channel_val ||
+            paranet_out_channel_dim.is_dynamic() || paranet_out_channel_dim != channel_val)
+            return false;
+
+        auto new_shape = ov::Shape(parent->get_input_partial_shape(0).size(), 1);
+
+        new_shape[channel_idx] = const_shape[channel_idx];
+        auto old_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(eltwise->get_input_node_shared_ptr(1));
+        auto new_const = std::make_shared<ov::op::v0::Constant>(*old_const, new_shape);
+        ov::replace_node_update_name(old_const, new_const);
+        ov::replace_output_update_name(eltwise->output(0), eltwise->input_value(0));
```

Comment:
this is eltwise operation, so it's not safe to use direct index here
because MatherPass automatically matches both options: BinaryElementwiseArithmetic(eltw_data_flow_in, eltw_const_in) and BinaryElementwiseArithmetic(eltw_const_in, eltw_data_flow_in)

Reply:
Fixed